File: balancer/service/navigate/src/core/Kv.cc
#include "Kv.h"

#include "log/Log.h"
#include "tool/Util.h"

Kv::Kv()
	:	_navigate_key_timeout_us(0)
{

}

Kv::~Kv()
{

}

bool Kv::load_kv(std::map<std::string, ServiceConfig::KV>& kv_map)
{
	bool b = true;

	get_str(kv_map, KvKey::navigate_key, _navigate_key, b);
	get_uint(kv_map, KvKey::navigate_key_timeout_us, _navigate_key_timeout_us, b);

	return b;
}

void Kv::get_str(std::map<std::string, ServiceConfig::KV>& kv_map,
				 const std::string& key, 
				 std::string& str,
				 bool& b)
{
	auto it = kv_map.find(key);
	if(it == kv_map.end())
	{
		b = false;
		B_LOG_ERROR << "kv str, key is not find, key=" << key;
		return;
	}

	if(it->second.val.empty())
	{
		b = false;
		B_LOG_ERROR << "kv str, val is empty, key=" << key;
		return;
	}

	str = it->second.val;
	B_LOG_ERROR << "kv str, success, key=" << key << ", val=" << it->second.val << ", str=" << str;
}

void Kv::get_uint(std::map<std::string, ServiceConfig::KV>& kv_map,
				  const std::string& key, 
				  unsigned int& i,
				  bool& b)
{
	auto it = kv_map.find(key);
	if(it == kv_map.end())
	{
		b = false;
		B_LOG_ERROR << "kv uint, key is not find, key=" << key;
		return;
	}

	if(it->second.val.empty())
	{
		b = false;
		B_LOG_ERROR << "kv uint, val is empty, key=" << key;
		return;
	}

	for(auto c :  it->second.val)
	{
		if(c < '0' || c > '9')
		{
			b = false;
			B_LOG_ERROR << "kv uint, val is not number, key=" << key << ", val=" << it->second.val;
			return;
		}
	}

	i = Util::str_2_uint(it->second.val);
	B_LOG_ERROR << "kv uint, success, key=" << key << ", val=" << it->second.val << ", i=" << i;
}

File: balancer/service/proxy/src/core/kv.py
# -*- coding: utf-8 -*-


class Kv:

    timeout = 'timeout'
    env = 'env'

    def __init__(self):
        self.timeout = 0
        self.env = ''

    def load_kv(self, kv_map):
        b = True

        self.get_uint(kv_map, Kv.timeout, self.timeout, b)
        self.get_str(kv_map, Kv.env, self.env, b)

        return b

    def get_str(self, kv_map, key, str, b):
        val = kv_map.get(key)
        if val is None:
            b = False
            return

        if len(val) == 0:
            b = False
            return

        str = val

    def get_uint(self, kv_map, key, i, b):
        val = kv_map.get(key)
        if val is None:
            b = False
            return

        if len(val) == 0:
            b = False
            return

        for s in val:
            if s < '0' or s > '9':
                b = False
                return

        i = int(val)




